src/MusicalScore/Graphical/VexFlow/VexFlowMusicSheetCalculator.ts

```diff
@@ -144,20 +144,35 @@ export class VexFlowMusicSheetCalculator extends MusicSheetCalculator {
   public calculateMeasureWidthFromLyrics(measuresVertical: GraphicalMeasure[], oldMinimumStaffEntriesWidth: number): number {
     let lastLyricsLabelHalfWidth: number = 0; // TODO lyrics entries may not be ordered correctly, create a dictionary
     let lastStaffEntryXPosition: number = 0;
+    // let lastLyricsEntryText: string = ""; // for easier debug purposes
+    // let lyricsEntryText: string = ""; // for easier debug purposes
+    // let lyricExtend: boolean = false; // for easier debug purposes
     let elongationFactorMeasureWidth: number = 1;
     for (const measure of measuresVertical) {
-      for (let i: number = 0; i < measure.staffEntries.length; i++) {
-        const staffEntry: GraphicalStaffEntry = measure.staffEntries[i];
+      // sort staffEntries by position, so that we always compare neighboring staffEntries
+      const staffEntriesSorted: GraphicalStaffEntry[] = measure.staffEntries.sort(
```

Comment:
Have you tried, that it doesn't work without sorting? They should be sorted actually already..

Reply:
Yes, i checked that with a debugging session, they can be unsorted. For example in An die Musik, for "war-mer Lieb' ent-zun-den", i had the StaffEntries with lyricEntries ["war","mer","ent","Lieb'"] and so on.
I also thought they should be sorted, because that's done somewhere in the code, i'd have to look where that is. But i'm pretty sure the entries weren't sorted, and that's why i got much better spacing with much less minDistance (0.3 -> 0.1) after sorting.

Reply:
interesting... maybe that's the reason for other problems. Let's keep an eye on it..